manager/integration/tests/common.py

```diff
@@ -1950,10 +1980,19 @@ def crash_replica_processes(client, api, volname, replicas=None,
                        "' | grep -v grep | awk '{print $2}'`"
         exec_instance_manager(api, r.instanceManagerName, kill_command)
 
-        # FIXME: this is a work around, should check this in-parallel.
-        #        https://github.com/longhorn/longhorn/issues/4045
+        # Check all replica status in-parallel but it might not make sure that
+        # replica is failed after checking and quickly recovering.
+        # https://github.com/longhorn/longhorn/issues/4045
         if wait_to_fail is True:
-            wait_for_replica_failed(client, volname, r['name'])
+            check_t = replicaThread(target=wait_for_replica_failed,
+                                    args=(client, volname, r['name'],
+                                          RETRY_COUNTS*2, RETRY_INTERVAL/2))
+            threads.append(check_t)
+            check_t.start()
+
+    if wait_to_fail:
+        for t in threads:
+            t.join()
```

Comment:
Suggest just raise an accumulated or one except when all threads finished instead of leaving some unhandled threads behind. This way can help us find some issues earlier like generally orphaned threads.

Reply:
Agree, I will modify codes for this.